### ISSUE TITLE
ansible-galaxy paging error

### ISSUE THREAD

**Issue by chouseknecht** [ansible/ansible#14715]
##### Issue Type:
- Bug Report
##### Summary:

https://github.com/ansible/galaxy-issues/issues/134. There is a paging bug in ansible-galaxy CLI. Some time ago when Django DRF was upgraded the `next` and `previous` values in the API changed. They began to include the protocol and server. They should not. At the same time ansible-galaxy CLI v2 changed so that it was erroneously concatenating `base_url` with the value found in `next`. The value of `base_url` unfortunately includes server, protocol and `/api/v1`.  

To get v2 working immediately the API has changed to drop the protocol, server and `api/v1` from `next`. It works. We added `next_link` which includes `/api/v1` as it should. 

v1.9.4 is broken. It expects Next to include `/api/v1`. 

Going forward both v2 and v1.94 should use `next_link` and `previous_link`, and `next` and `previous` should be deprecated. 
##### Steps To Reproduce:

```
ansible-galaxy install geerlingguy.php -vvv
```
##### Expected Results:

It should install the role successfully.
##### Actual Results:

```
No config file found; using defaults
Opened /Users/cindy/.ansible_galaxy
Validate TLS certificates: True
Connecting to galaxy_server: https://galaxy.ansible.com
https://galaxy.ansible.com/api/
Base API: https://galaxy.ansible.com/api/v1
Installing role geerlingguy.php
Opened /Users/cindy/.ansible_galaxy
Validate TLS certificates: True
Connecting to galaxy_server: https://galaxy.ansible.com
https://galaxy.ansible.com/api/
Base API: https://galaxy.ansible.com/api/v1
- downloading role 'php', owned by geerlingguy
https://galaxy.ansible.com/api/v1/roles/?owner__username=geerlingguy&name=php
https://galaxy.ansible.com/api/v1/roles/432/versions/?page_size=50
https://galaxy.ansible.com/api/v1https://galaxy.ansible.com/api/v1/roles/432/versions/?page=2&page_size=50
Unexpected Exception: object of type 'NoneType' has no len()
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/2.0.0.2/libexec/bin/ansible-galaxy", line 85, in <module>
    sys.exit(cli.run())
  File "/usr/local/Cellar/ansible/2.0.0.2/libexec/lib/python2.7/site-packages/ansible/cli/galaxy.py", line 152, in run
    self.execute()
  File "/usr/local/Cellar/ansible/2.0.0.2/libexec/lib/python2.7/site-packages/ansible/cli/__init__.py", line 99, in execute
    fn()
  File "/usr/local/Cellar/ansible/2.0.0.2/libexec/lib/python2.7/site-packages/ansible/cli/galaxy.py", line 399, in execute_install
    installed = role.install()
  File "/usr/local/Cellar/ansible/2.0.0.2/libexec/lib/python2.7/site-packages/ansible/galaxy/role.py", line 212, in install
    if len(role_versions) > 0:
TypeError: object of type 'NoneType' has no len()
```

**Comment by chouseknecht** [comment]
https://github.com/ansible/ansible/pull/14713

https://github.com/ansible/ansible/pull/14712

**Comment by bcoca** [comment]
just to point out, this does not fix 1.9.4 or other versions that have the incorrect path to galaxy, those will continue to error out.

Closing this ticket as per the merged PRs above
